File: banco_problemas_aplicados/nivel_intermedio/soluciones/cpp/problema25.cpp
/*
 * Problema 25 - Particion Pares/Impares en Vector
 * Primero los valores de indices pares, luego los de indices impares, en un vector auxiliar.
 */
#include <iostream>
using namespace std;

void particiona(int arr[], int tam, int res[]) {
    int k = 0;
    for (int i = 0; i < tam; i++) {
        if (i % 2 == 0) {
            res[k] = arr[i];
            k = k + 1;
        }
    }
    for (int i = 0; i < tam; i++) {
        if (i % 2 != 0) {
            res[k] = arr[i];
            k = k + 1;
        }
    }
}

int main() {
    int tam;
    cout << "Ingrese el tamano del vector: ";
    cin >> tam;

    int* arr = new int[tam];
    int* res = new int[tam];
    for (int i = 0; i < tam; i++) {
        cout << "Dato " << i << ": ";
        cin >> arr[i];
    }

    particiona(arr, tam, res);

    cout << "Vector particionado (pares | impares): ";
    for (int i = 0; i < tam; i++) {
        cout << res[i] << " ";
    }
    cout << endl;

    delete[] arr;
    delete[] res;
    return 0;
}

File: banco_problemas_aplicados/nivel_intermedio/soluciones/java/Problema25.java
/*
 * Problema 25 - Particion Pares/Impares en Vector
 * Primero los valores de indices pares, luego los de indices impares, en un arreglo auxiliar.
 */
import java.util.Scanner;

public class Problema25 {

    static void particiona(int[] arr, int tam, int[] res) {
        int k = 0;
        for (int i = 0; i < tam; i++) {
            if (i % 2 == 0) {
                res[k] = arr[i];
                k = k + 1;
            }
        }
        for (int i = 0; i < tam; i++) {
            if (i % 2 != 0) {
                res[k] = arr[i];
                k = k + 1;
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el tamano del vector: ");
        int tam = sc.nextInt();

        int[] arr = new int[tam];
        int[] res = new int[tam];
        for (int i = 0; i < tam; i++) {
            System.out.print("Dato " + i + ": ");
            arr[i] = sc.nextInt();
        }

        particiona(arr, tam, res);

        System.out.print("Vector particionado (pares | impares): ");
        for (int i = 0; i < tam; i++) {
            System.out.print(res[i] + " ");
        }
        System.out.println();
        sc.close();
    }
}
